### ISSUE TITLE
how to pass x-b3-* headers into ext authz grpc server

### ISSUE THREAD

**Issue by vgsub** [envoyproxy/envoy#8914]
Hi guys, I'm trying to pass zipkin trace and span Ids into my java based ext auth server, but it seems like it cannot be done. Here is my config:

```
node:
  id: ${HOSTNAME}
admin:
  access_log_path: /dev/stdout
  address:
    socket_address:
      address: 0.0.0.0
      port_value: 9901
dynamic_resources:
  ads_config:
    api_type: GRPC
    grpc_services:
      envoy_grpc:
        cluster_name: xds_cluster
  cds_config:
    ads: {}
tracing:
  http:
    name: envoy.zipkin
    typed_config:
      "@type": type.googleapis.com/envoy.config.trace.v2.ZipkinConfig
      collector_cluster: zipkin_cluster
      collector_endpoint: "/api/v1/spans"
static_resources:
  listeners:
    - name: main
      address:
        socket_address:
          address: 0.0.0.0
          port_value: 10000
      filter_chains:
        - filters:
            - name: envoy.http_connection_manager
              typed_config:
                "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
                stat_prefix: ingress_http
                add_user_agent: true
                generate_request_id: true
                tracing:
                  operation_name: ingress
                  request_headers_for_tags: X-Real-IP
                access_log:
                  - name: envoy.file_access_log
                    filter:
                      or_filter:
                        filters:
                          - status_code_filter:
                              comparison:
                                op: GE
                                value:
                                  default_value: 0
                                  runtime_key: "null"
                          - duration_filter:
                              comparison:
                                op: GE
                                value:
                                  default_value: 0
                                  runtime_key: "null"
                    config:
                      json_format:
                        timestamp: "%START_TIME%"
                        microsecond: "%START_TIME(%6f)%"
                        X-Real-IP: "%REQ(X-Real-IP)%"
                        user: "%REQ(tcrm-user)%"
                        X-B3-SpanId: "%REQ(X-B3-SpanId)%"
                        X-B3-TraceId: "%REQ(X-B3-TraceId)%"
                        X-B3-ParentSpanId: "%REQ(X-B3-ParentSpanId)%"
                        envoy.request_method: "%REQ(:METHOD)%"
                        envoy.path: "%REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%"
                        envoy.protocol: "%PROTOCOL%"
                        envoy.response_code: "%RESPONSE_CODE%"
                        envoy.response_flags: "%RESPONSE_FLAGS%"
                        envoy.recived_bytes: "%BYTES_RECEIVED%"
                        envoy.send_bytes: "%BYTES_SENT%"
                        envoy.duration: "%DURATION%"
                        envoy.response_duration: "%RESPONSE_DURATION%"
                        envoy.response_tx_duration: "%RESPONSE_TX_DURATION%"
                        envoy.response_time: "%RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%"
                        envoy.request_user_agent: "%REQ(USER-AGENT)%"
                        envoy.upstream_host: "%UPSTREAM_HOST%"
                        logger_name: "ACCESS_LOG"
                      path: /dev/stdout
                route_config:
                  name: local_route
                  virtual_hosts:
                    - name: local_service
                      domains: ["*"]
                      routes:
                        - match: { prefix: "/" }
                          route: { host_rewrite: www.google.com, cluster: service_google }
                http_filters:
                  - name: envoy.lua
                    typed_config:
                      "@type": type.googleapis.com/envoy.config.filter.http.lua.v2.Lua
                      inline_code: %some code%                        
                  - name: envoy.cors
                  - name: envoy.filters.http.jwt_authn
                    config:
                      providers:
                        local_provider:
                          forward: true
                          local_jwks:
                            #  /oauth/token_key converted PEM-to-JWK
                            inline_string: %jwt%
                      rules:                       
                        - match:
                            prefix: "/"
                          requires: { provider_name: local_provider }
                  - name: envoy.ext_authz
                    typed_config:
                      "@type": type.googleapis.com/envoy.config.filter.http.ext_authz.v2.ExtAuthz
                      grpc_service:
                        initial_metadata:
                          - key: x-b3-spanid
                            value: "%REQ(X-B3-SpanId)%"
                          - key: x-b3-traceid
                            value: "%REQ(X-B3-TraceId)%"
                        envoy_grpc:
                          cluster_name: ext-authz
                  - name: envoy.router
  clusters:
    - name: xds_cluster
      connect_timeout: 3s
      type: STRICT_DNS
      dns_lookup_family: V4_ONLY
      http2_protocol_options: {}
      lb_policy: ROUND_ROBIN
      load_assignment:
        cluster_name: xds_cluster
        endpoints:
          - lb_endpoints:
              - endpoint:
                  address:
                    socket_address:
                      address: docker.for.mac.localhost
                      port_value: 8888
    - name: ext-authz
      connect_timeout: 3s
      type: STRICT_DNS
      dns_lookup_family: V4_ONLY
      http2_protocol_options: {}
      lb_policy: ROUND_ROBIN
      health_checks:
        timeout: 5s
        interval: 5s
        no_traffic_interval: 5s
        unhealthy_threshold: 2
        healthy_threshold: 3
        event_log_path: /dev/stdout
        grpc_health_check: {}
      load_assignment:
        cluster_name: acl_cluster
        endpoints:
          - lb_endpoints:
              - endpoint:
                  address:
                    socket_address:
                      address: docker.for.mac.localhost
                      port_value: 8889
    - name: zipkin_cluster
      connect_timeout: 3s
      type: STRICT_DNS
      dns_lookup_family: V4_ONLY
      http2_protocol_options: {}
      lb_policy: ROUND_ROBIN
      load_assignment:
        cluster_name: zipkin_cluster
        endpoints:
          - lb_endpoints:
              - endpoint:
                  address:
                    socket_address:
                      address: docker.for.mac.localhost
                      port_value: 9411
    - name: service_google
      connect_timeout: 0.25s
      type: LOGICAL_DNS
      dns_lookup_family: V4_ONLY
      lb_policy: ROUND_ROBIN
      load_assignment:
        cluster_name: service_google
        endpoints:
          - lb_endpoints:
               - endpoint:
                    address:
                      socket_address:
                        address: www.google.com
                        port_value: 443
      tls_context:
        sni: www.google.com
```

Here is the log of a test request:

```
[2019-11-06 09:16:04.534][15][trace][connection] [source/common/network/connection_impl.cc:462] [C6] socket event: 3
[2019-11-06 09:16:04.534][15][trace][connection] [source/common/network/connection_impl.cc:550] [C6] write ready
[2019-11-06 09:16:04.534][15][trace][connection] [source/common/network/connection_impl.cc:500] [C6] read ready
[2019-11-06 09:16:04.534][15][trace][connection] [source/common/network/raw_buffer_socket.cc:24] [C6] read returns: 1820
[2019-11-06 09:16:04.534][15][trace][connection] [source/common/network/raw_buffer_socket.cc:38] [C6] read error: Resource temporarily unavailable
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:409] [C6] parsing 1820 bytes
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:559] [C6] message begin
[2019-11-06 09:16:04.534][15][debug][http] [source/common/http/conn_manager_impl.cc:259] [C6] new stream
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Content-Type value=application/json
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Authorization value=Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX25hbWUiOiJ2Lmcuc3ViYm90aW4iLCJmdWxsTmFtZSI6ItCh0YPQsdCx0L7RgtC40L0g0JLQsNC00LjQvCDQk9C10L3QvdCw0LTRjNC10LLQuNGHIiwiZW1wbG95ZWVJZCI6IjMtMTNETFRNOFg3IiwidXNlcklkIjoiOTc4MmY5NDctOGZmZS00ZmU1LTlmNjMtMmNjZTJiNDViZTM0IiwiYXV0aG9yaXRpZXMiOlsiVENSTSBVc2VyIiwiVENSTV9FTVBMT1lFRVMiLCJST0xFX1VTRVIiLCJJTU1PUlRBTCIsIlJPTEVfQURNSU4iLCJUQ1JNX1BST0NfTUFOQUdFUiIsIlRDUk1fVElNRUxJTkVfQ09ORklHX0FETUlOIiwiUk9MRV9PRkZJQ0VfVVNFUiJdLCJjbGllbnRfaWQiOiJ3ZWJfYXBwIiwiZmlyc3ROYW1lIjoi0JLQsNC00LjQvCIsInRlbFR5cGUiOm51bGwsImdyYW50X3R5cGUiOiJwYXNzd29yZCIsInN1cm5hbWUiOiLQodGD0LHQsdC-0YLQuNC9Iiwic2NvcGUiOlsib3BlbmlkIl0sInBvc2l0aW9uIjoi0KHRgtCw0YDRiNC40Lkg0YDQsNC30YDQsNCx0L7RgtGH0LjQuiIsInVzZXJUeXBlIjoiT0ZGSUNFIiwiZXhwIjoxNTczMDQ2NjMzLCJkZXBhcnRtZW50Ijoi0J7RgtC00LXQuyDRgNCw0LfRgNCw0LHQvtGC0LrQuCDQstC90YPRgtGA0LXQvdC90LjRhSDQuNC90YLQtdGA0YTQtdC50YHQvtCyIiwic2llYmVsTG9naW4iOiJWR1NVQkJPVElOIiwianRpIjoiZmVhZjY3ZTItZjNjOC00MjAxLThhNmQtM2I3MGRkMTZhNmFiIn0.SK03pqaH72acHWHk08WKEsIcSszHGFkLJd0rI-PVlHmrm7s7Fki97Uea4Nnj7GEY_FPI-ZGiGX_QZ0r_RIQDxrXZPoFCU-c0jnOQr0b9hyhm2oy0F3oi9i5q87HVm0M0IQwagXmbufgG2RBwaTv8L2F93b2YbRC_XDR7LOWpYM3o2fLzwxP89lqyHgovbo7obsFJNiyOnIKorvnqprF4-s9sxybWsW7alwvr7KfPQkAx6WeqG9889YftFxY6aez2T2Ol9Lz9kc4fhK08kurMcc7dWJdjkZYG255NhYJLYZJO0RMyWncFwo1PcwlyhMvobYI4Wq6tkSDIvxRb4Ppwcg
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=User-Agent value=PostmanRuntime/7.17.1
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Accept value=*/*
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Cache-Control value=no-cache
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Postman-Token value=8fdf9633-348b-403e-8565-f17fae171c90
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Host value=localhost:10000
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Accept-Encoding value=gzip, deflate
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Content-Length value=193
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:501] [C6] headers complete
[2019-11-06 09:16:04.534][15][trace][http] [source/common/http/http1/codec_impl.cc:371] [C6] completed header: key=Connection value=keep-alive
[2019-11-06 09:16:04.534][15][debug][http] [source/common/http/conn_manager_impl.cc:708] [C6][S15852472204836360053] request headers complete (end_stream=false):
':authority', 'localhost:10000'
':path', '/api/v1/test'
':method', 'POST'
'content-type', 'application/json'
'authorization', 'Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX25hbWUiOiJ2Lmcuc3ViYm90aW4iLCJmdWxsTmFtZSI6ItCh0YPQsdCx0L7RgtC40L0g0JLQsNC00LjQvCDQk9C10L3QvdCw0LTRjNC10LLQuNGHIiwiZW1wbG95ZWVJZCI6IjMtMTNETFRNOFg3IiwidXNlcklkIjoiOTc4MmY5NDctOGZmZS00ZmU1LTlmNjMtMmNjZTJiNDViZTM0IiwiYXV0aG9yaXRpZXMiOlsiVENSTSBVc2VyIiwiVENSTV9FTVBMT1lFRVMiLCJST0xFX1VTRVIiLCJJTU1PUlRBTCIsIlJPTEVfQURNSU4iLCJUQ1JNX1BST0NfTUFOQUdFUiIsIlRDUk1fVElNRUxJTkVfQ09ORklHX0FETUlOIiwiUk9MRV9PRkZJQ0VfVVNFUiJdLCJjbGllbnRfaWQiOiJ3ZWJfYXBwIiwiZmlyc3ROYW1lIjoi0JLQsNC00LjQvCIsInRlbFR5cGUiOm51bGwsImdyYW50X3R5cGUiOiJwYXNzd29yZCIsInN1cm5hbWUiOiLQodGD0LHQsdC-0YLQuNC9Iiwic2NvcGUiOlsib3BlbmlkIl0sInBvc2l0aW9uIjoi0KHRgtCw0YDRiNC40Lkg0YDQsNC30YDQsNCx0L7RgtGH0LjQuiIsInVzZXJUeXBlIjoiT0ZGSUNFIiwiZXhwIjoxNTczMDQ2NjMzLCJkZXBhcnRtZW50Ijoi0J7RgtC00LXQuyDRgNCw0LfRgNCw0LHQvtGC0LrQuCDQstC90YPRgtGA0LXQvdC90LjRhSDQuNC90YLQtdGA0YTQtdC50YHQvtCyIiwic2llYmVsTG9naW4iOiJWR1NVQkJPVElOIiwianRpIjoiZmVhZjY3ZTItZjNjOC00MjAxLThhNmQtM2I3MGRkMTZhNmFiIn0.SK03pqaH72acHWHk08WKEsIcSszHGFkLJd0rI-PVlHmrm7s7Fki97Uea4Nnj7GEY_FPI-ZGiGX_QZ0r_RIQDxrXZPoFCU-c0jnOQr0b9hyhm2oy0F3oi9i5q87HVm0M0IQwagXmbufgG2RBwaTv8L2F93b2YbRC_XDR7LOWpYM3o2fLzwxP89lqyHgovbo7obsFJNiyOnIKorvnqprF4-s9sxybWsW7alwvr7KfPQkAx6WeqG9889YftFxY6aez2T2Ol9Lz9kc4fhK08kurMcc7dWJdjkZYG255NhYJLYZJO0RMyWncFwo1PcwlyhMvobYI4Wq6tkSDIvxRb4Ppwcg'
'user-agent', 'PostmanRuntime/7.17.1'
'accept', '*/*'
'cache-control', 'no-cache'
'postman-token', '8fdf9633-348b-403e-8565-f17fae171c90'
'accept-encoding', 'gzip, deflate'
'content-length', '193'
'connection', 'keep-alive'

[2019-11-06 09:16:04.535][15][debug][jwt] [source/extensions/filters/http/jwt_authn/filter.cc:101] Called Filter : setDecoderFilterCallbacks
[2019-11-06 09:16:04.535][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:122] creating N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78
[2019-11-06 09:16:04.535][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:122] creating N5Envoy10Extensions11HttpFilters3Lua16HeaderMapWrapperE at 0x415d5148
[2019-11-06 09:16:04.535][15][debug][lua] [source/extensions/filters/http/lua/lua_filter.cc:589] script log: path: /api/v1/test
[2019-11-06 09:16:04.535][15][debug][lua] [source/extensions/filters/http/lua/lua_filter.cc:589] script log: !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
[2019-11-06 09:16:04.535][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:122] creating N5Envoy10Extensions11HttpFilters3Lua17HeaderMapIteratorE at 0x415d51f0
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key::authority <--> value:localhost:10000
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key::path <--> value:/api/v1/test
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key::method <--> value:POST
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:content-type <--> value:application/json
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:authorization <--> value:Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VyX25hbWUiOiJ2Lmcuc3ViYm90aW4iLCJmdWxsTmFtZSI6ItCh0YPQsdCx0L7RgtC40L0g0JLQsNC00LjQvCDQk9C10L3QvdCw0LTRjNC10LLQuNGHIiwiZW1wbG95ZWVJZCI6IjMtMTNETFRNOFg3IiwidXNlcklkIjoiOTc4MmY5NDctOGZmZS00ZmU1LTlmNjMtMmNjZTJiNDViZTM0IiwiYXV0aG9yaXRpZXMiOlsiVENSTSBVc2VyIiwiVENSTV9FTVBMT1lFRVMiLCJST0xFX1VTRVIiLCJJTU1PUlRBTCIsIlJPTEVfQURNSU4iLCJUQ1JNX1BST0NfTUFOQUdFUiIsIlRDUk1fVElNRUxJTkVfQ09ORklHX0FETUlOIiwiUk9MRV9PRkZJQ0VfVVNFUiJdLCJjbGllbnRfaWQiOiJ3ZWJfYXBwIiwiZmlyc3ROYW1lIjoi0JLQsNC00LjQvCIsInRlbFR5cGUiOm51bGwsImdyYW50X3R5cGUiOiJwYXNzd29yZCIsInN1cm5hbWUiOiLQodGD0LHQsdC-0YLQuNC9Iiwic2NvcGUiOlsib3BlbmlkIl0sInBvc2l0aW9uIjoi0KHRgtCw0YDRiNC40Lkg0YDQsNC30YDQsNCx0L7RgtGH0LjQuiIsInVzZXJUeXBlIjoiT0ZGSUNFIiwiZXhwIjoxNTczMDQ2NjMzLCJkZXBhcnRtZW50Ijoi0J7RgtC00LXQuyDRgNCw0LfRgNCw0LHQvtGC0LrQuCDQstC90YPRgtGA0LXQvdC90LjRhSDQuNC90YLQtdGA0YTQtdC50YHQvtCyIiwic2llYmVsTG9naW4iOiJWR1NVQkJPVElOIiwianRpIjoiZmVhZjY3ZTItZjNjOC00MjAxLThhNmQtM2I3MGRkMTZhNmFiIn0.SK03pqaH72acHWHk08WKEsIcSszHGFkLJd0rI-PVlHmrm7s7Fki97Uea4Nnj7GEY_FPI-ZGiGX_QZ0r_RIQDxrXZPoFCU-c0jnOQr0b9hyhm2oy0F3oi9i5q87HVm0M0IQwagXmbufgG2RBwaTv8L2F93b2YbRC_XDR7LOWpYM3o2fLzwxP89lqyHgovbo7obsFJNiyOnIKorvnqprF4-s9sxybWsW7alwvr7KfPQkAx6WeqG9889YftFxY6aez2T2Ol9Lz9kc4fhK08kurMcc7dWJdjkZYG255NhYJLYZJO0RMyWncFwo1PcwlyhMvobYI4Wq6tkSDIvxRb4Ppwcg
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:user-agent <--> value:PostmanRuntime/7.17.1
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:accept <--> value:*/*
[2019-11-06 09:16:04.535][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:cache-control <--> value:no-cache
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:postman-token <--> value:8fdf9633-348b-403e-8565-f17fae171c90
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:accept-encoding <--> value:gzip, deflate
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:content-length <--> value:193
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:x-forwarded-proto <--> value:http
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:x-envoy-downstream-service-cluster <--> value:api-gateway
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:x-envoy-downstream-service-node <--> value:${HOSTNAME}
[2019-11-06 09:16:04.536][15][warning][lua] [source/extensions/filters/http/lua/lua_filter.cc:595] script log: key:x-request-id <--> value:9c28a366-29f5-9e2a-b6d7-e9bf41052db6
[2019-11-06 09:16:04.536][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions11HttpFilters3Lua17HeaderMapIteratorE at 0x415d51f0
[2019-11-06 09:16:04.536][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:122] creating N5Envoy10Extensions7Filters6Common3Lua18MetadataMapWrapperE at 0x415d37c0
[2019-11-06 09:16:04.536][15][debug][lua] [source/extensions/filters/http/lua/lua_filter.cc:589] script log: not found metadata default_cluster
[2019-11-06 09:16:04.536][15][debug][lua] [source/extensions/filters/common/lua/lua.cc:37] coroutine finished
[2019-11-06 09:16:04.536][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78
[2019-11-06 09:16:04.536][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions11HttpFilters3Lua16HeaderMapWrapperE at 0x415d5148
[2019-11-06 09:16:04.536][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions7Filters6Common3Lua18MetadataMapWrapperE at 0x415d37c0
[2019-11-06 09:16:04.536][15][trace][http] [source/common/http/conn_manager_impl.cc:957] [C6][S15852472204836360053] decode headers called: filter=0x56520bda2780 status=0
[2019-11-06 09:16:04.536][15][trace][http] [source/common/http/conn_manager_impl.cc:957] [C6][S15852472204836360053] decode headers called: filter=0x56520b98ee40 status=0
[2019-11-06 09:16:04.536][15][debug][jwt] [source/extensions/filters/http/jwt_authn/filter.cc:33] Called Filter : decodeHeaders
[2019-11-06 09:16:04.536][15][debug][jwt] [source/extensions/filters/http/jwt_authn/matcher.cc:68] Prefix requirement '/' matched.
[2019-11-06 09:16:04.536][15][debug][jwt] [source/extensions/filters/http/jwt_authn/authenticator.cc:107] JWT authentication starts (allow_failed=false)
[2019-11-06 09:16:04.537][15][debug][jwt] [source/extensions/filters/http/jwt_authn/authenticator.cc:127] Verifying JWT token of issuer 
[2019-11-06 09:16:04.537][15][debug][jwt] [source/extensions/filters/http/jwt_authn/authenticator.cc:245] JWT token verification completed with: OK
[2019-11-06 09:16:04.537][15][debug][jwt] [source/extensions/filters/http/jwt_authn/filter.cc:61] Called Filter : check complete OK
[2019-11-06 09:16:04.537][15][trace][http] [source/common/http/conn_manager_impl.cc:957] [C6][S15852472204836360053] decode headers called: filter=0x56520bda2600 status=0
[2019-11-06 09:16:04.537][15][trace][filter] [source/extensions/filters/http/ext_authz/ext_authz.cc:80] [C6][S15852472204836360053] ext_authz filter calling authorization server
[2019-11-06 09:16:04.537][15][debug][router] [source/common/router/router.cc:434] [C0][S2826641673151063479] cluster 'ext-authz' match for URL '/envoy.service.auth.v2.Authorization/Check'
[2019-11-06 09:16:04.538][15][debug][router] [source/common/router/router.cc:549] [C0][S2826641673151063479] router decoding headers:
':method', 'POST'
':path', '/envoy.service.auth.v2.Authorization/Check'
':authority', 'ext-authz'
':scheme', 'http'
'te', 'trailers'
'grpc-timeout', '200m'
'content-type', 'application/grpc'
'x-b3-traceid', 'd04e3d2f5120f982'
'x-b3-spanid', '37b1e99dd69816d6'
'x-b3-parentspanid', 'd04e3d2f5120f982'
'x-b3-sampled', '1'
'x-envoy-internal', 'true'
'x-forwarded-for', '172.17.0.2'
'x-envoy-expected-rq-timeout-ms', '200'

[2019-11-06 09:16:04.538][15][debug][pool] [source/common/http/http2/conn_pool.cc:98] [C7] creating stream
[2019-11-06 09:16:04.538][15][debug][router] [source/common/router/router.cc:1618] [C0][S2826641673151063479] pool ready
[2019-11-06 09:16:04.538][15][trace][http2] [source/common/http/http2/codec_impl.cc:666] [C7] about to send frame type=1, flags=4
[2019-11-06 09:16:04.538][15][trace][http2] [source/common/http/http2/codec_impl.cc:720] [C7] send data: bytes=96
[2019-11-06 09:16:04.538][15][trace][connection] [source/common/network/connection_impl.cc:398] [C7] writing 96 bytes, end_stream false
[2019-11-06 09:16:04.538][15][trace][http2] [source/common/http/http2/codec_impl.cc:608] [C7] sent frame type=1
[2019-11-06 09:16:04.538][15][trace][router] [source/common/router/router.cc:1487] [C0][S2826641673151063479] proxying 1960 bytes
[2019-11-06 09:16:04.538][15][trace][connection] [source/common/network/connection_impl.cc:398] [C7] writing 1969 bytes, end_stream false
[2019-11-06 09:16:04.538][15][trace][http2] [source/common/http/http2/codec_impl.cc:608] [C7] sent frame type=0
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/conn_manager_impl.cc:957] [C6][S15852472204836360053] decode headers called: filter=0x56520bda2960 status=4
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/http1/codec_impl.cc:696] [C6] body size=193
[2019-11-06 09:16:04.539][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:192] marking live N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78
[2019-11-06 09:16:04.539][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/conn_manager_impl.cc:1096] [C6][S15852472204836360053] decode data called: filter=0x56520bda2780 status=0
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/conn_manager_impl.cc:1096] [C6][S15852472204836360053] decode data called: filter=0x56520b98ee40 status=0
[2019-11-06 09:16:04.539][15][debug][jwt] [source/extensions/filters/http/jwt_authn/filter.cc:85] Called Filter : decodeData
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/conn_manager_impl.cc:1096] [C6][S15852472204836360053] decode data called: filter=0x56520bda2600 status=0
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/http1/codec_impl.cc:546] [C6] message complete
[2019-11-06 09:16:04.539][15][debug][http] [source/common/http/conn_manager_impl.cc:1248] [C6][S15852472204836360053] request end stream
[2019-11-06 09:16:04.539][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:192] marking live N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78
[2019-11-06 09:16:04.539][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/conn_manager_impl.cc:1096] [C6][S15852472204836360053] decode data called: filter=0x56520bda2780 status=0
[2019-11-06 09:16:04.539][15][trace][http] [source/common/http/conn_manager_impl.cc:1096] [C6][S15852472204836360053] decode data called: filter=0x56520b98ee40 status=0
[2019-11-06 09:16:04.539][15][debug][jwt] [source/extensions/filters/http/jwt_authn/filter.cc:85] Called Filter : decodeData
[2019-11-06 09:16:04.540][15][trace][http] [source/common/http/conn_manager_impl.cc:1096] [C6][S15852472204836360053] decode data called: filter=0x56520bda2600 status=0
[2019-11-06 09:16:04.540][15][trace][http] [source/common/http/http1/codec_impl.cc:430] [C6] parsed 1820 bytes
[2019-11-06 09:16:04.540][15][trace][connection] [source/common/network/connection_impl.cc:293] [C6] readDisable: enabled=true disable=true
[2019-11-06 09:16:04.540][15][trace][connection] [source/common/network/connection_impl.cc:462] [C7] socket event: 2
[2019-11-06 09:16:04.540][15][trace][connection] [source/common/network/connection_impl.cc:550] [C7] write ready
[2019-11-06 09:16:04.540][15][trace][connection] [source/common/network/raw_buffer_socket.cc:67] [C7] write returns: 2065
[2019-11-06 09:16:04.540][15][trace][connection] [source/common/network/connection_impl.cc:462] [C6] socket event: 2
[2019-11-06 09:16:04.540][15][trace][connection] [source/common/network/connection_impl.cc:550] [C6] write ready
[2019-11-06 09:16:04.544][15][trace][connection] [source/common/network/connection_impl.cc:462] [C7] socket event: 3
[2019-11-06 09:16:04.544][15][trace][connection] [source/common/network/connection_impl.cc:550] [C7] write ready
[2019-11-06 09:16:04.544][15][trace][connection] [source/common/network/connection_impl.cc:500] [C7] read ready
[2019-11-06 09:16:04.544][15][trace][connection] [source/common/network/raw_buffer_socket.cc:24] [C7] read returns: 58
[2019-11-06 09:16:04.544][15][trace][connection] [source/common/network/raw_buffer_socket.cc:38] [C7] read error: Resource temporarily unavailable
[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:405] [C7] dispatching 58 bytes
[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:468] [C7] about to recv frame type=1, flags=36
[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:483] [C7] recv frame type=1
[2019-11-06 09:16:04.544][15][debug][router] [source/common/router/router.cc:1036] [C0][S2826641673151063479] upstream headers complete: end_stream=false
[2019-11-06 09:16:04.544][15][debug][http] [source/common/http/async_client_impl.cc:93] async http request response headers (end_stream=false):
':status', '200'
'content-type', 'application/grpc'
'grpc-encoding', 'identity'
'grpc-accept-encoding', 'gzip'
'x-envoy-upstream-service-time', '5'

[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:468] [C7] about to recv frame type=0, flags=0
[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:483] [C7] recv frame type=0
[2019-11-06 09:16:04.544][15][trace][http] [source/common/http/async_client_impl.cc:109] async http request response data (length=16 end_stream=false)
[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:468] [C7] about to recv frame type=1, flags=37
[2019-11-06 09:16:04.544][15][trace][http2] [source/common/http/http2/codec_impl.cc:483] [C7] recv frame type=1
[2019-11-06 09:16:04.544][15][debug][client] [source/common/http/codec_client.cc:101] [C7] response complete
[2019-11-06 09:16:04.544][15][trace][main] [source/common/event/dispatcher_impl.cc:160] item added to deferred deletion list (size=1)
[2019-11-06 09:16:04.544][15][debug][pool] [source/common/http/http2/conn_pool.cc:236] [C7] destroying stream: 0 remaining
[2019-11-06 09:16:04.544][15][debug][http] [source/common/http/async_client_impl.cc:119] async http request response trailers:
'grpc-status', '0'

[2019-11-06 09:16:04.544][15][trace][filter] [source/extensions/filters/http/ext_authz/ext_authz.cc:193] [C6][S15852472204836360053] ext_authz filter rejected the request. Response status code: '403
[2019-11-06 09:16:04.545][15][debug][http] [source/common/http/conn_manager_impl.cc:1345] [C6][S15852472204836360053] Sending local reply with details ext_authz_denied
[2019-11-06 09:16:04.545][15][trace][filter] [source/extensions/filters/http/ext_authz/ext_authz.cc:217] [C6][S15852472204836360053] ext_authz filter added header(s) to the local response:
[2019-11-06 09:16:04.545][15][trace][http] [source/common/http/conn_manager_impl.cc:1432] [C6][S15852472204836360053] encode headers called: filter=0x56520bdf43c0 status=0
[2019-11-06 09:16:04.545][15][trace][http] [source/common/http/conn_manager_impl.cc:1432] [C6][S15852472204836360053] encode headers called: filter=0x56520bdf4820 status=0
[2019-11-06 09:16:04.545][15][trace][http] [source/common/http/conn_manager_impl.cc:1432] [C6][S15852472204836360053] encode headers called: filter=0x56520bdf48c0 status=0
[2019-11-06 09:16:04.545][15][debug][http] [source/common/http/conn_manager_impl.cc:1543] [C6][S15852472204836360053] encoding headers via codec (end_stream=true):
':status', '403'
'date', 'Wed, 06 Nov 2019 09:16:04 GMT'
'server', 'envoy'

[2019-11-06 09:16:04.545][15][trace][connection] [source/common/network/connection_impl.cc:398] [C6] writing 97 bytes, end_stream false
[2019-11-06 09:16:04.545][15][debug][jwt] [source/extensions/filters/http/jwt_authn/filter.cc:26] Called Filter : onDestroy
[2019-11-06 09:16:04.545][15][trace][main] [source/common/event/dispatcher_impl.cc:160] item added to deferred deletion list (size=2)
[2019-11-06 09:16:04.545][15][trace][connection] [source/common/network/connection_impl.cc:293] [C6] readDisable: enabled=false disable=false
[2019-11-06 09:16:04.545][15][trace][main] [source/common/event/dispatcher_impl.cc:160] item added to deferred deletion list (size=3)
[2019-11-06 09:16:04.545][15][trace][main] [source/common/event/dispatcher_impl.cc:160] item added to deferred deletion list (size=4)
[2019-11-06 09:16:04.545][15][debug][http2] [source/common/http/http2/codec_impl.cc:742] [C7] stream closed: 0
[2019-11-06 09:16:04.545][15][trace][main] [source/common/event/dispatcher_impl.cc:160] item added to deferred deletion list (size=5)
[2019-11-06 09:16:04.545][15][trace][http2] [source/common/http/http2/codec_impl.cc:424] [C7] dispatched 58 bytes
[2019-11-06 09:16:04.545][15][trace][main] [source/common/event/dispatcher_impl.cc:79] clearing deferred deletion list (size=5)
[2019-11-06 09:16:04.546][15][trace][lua] [bazel-out/k8-opt/bin/source/extensions/filters/common/lua/_virtual_includes/lua_lib/extensions/filters/common/lua/lua.h:183] marking dead N5Envoy10Extensions11HttpFilters3Lua19StreamHandleWrapperE at 0x415d2f78

```

gRPC metadata contains all needed headers but they are not accessible inside AuthorizarionGrpc service, #7272 is related to that. 

My question is: is there any way to fetch these headers or put them into CheckResponse object?

**Comment by vgsub** [comment]
Resolved by adding grpc opentracing + opentrace-brave (https://github.com/opentracing-contrib/java-grpc)
